dont-wreck-my-house/src/main/java/learn/lodging/domain/GuestService.java

```diff
@@ -0,0 +1,138 @@
+package learn.lodging.domain;
+
+import learn.lodging.data.DataException;
+import learn.lodging.data.GuestRepository;
+import learn.lodging.models.Guest;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Service
+public class GuestService {
+
+    private final GuestRepository repository;
+    private final ArrayList<String> ALL_STATES_ACRONYMS = new ArrayList<>(List.of("AK", "AL", "AR", "AS", "AZ", "CA", "CO",
+            "CT", "DC", "DE", "FL", "GA", "GU", "HI", "IA", "ID", "IL", "IN", "KS", "KY", "LA", "MA", "MD", "ME",
+            "MI", "MN", "MO", "MP", "MS", "MT", "NC", "ND", "NE", "NH", "NJ", "NM", "NV", "NY", "OH", "OK", "OR",
+            "PA", "PR", "RI", "SC", "SD", "TN", "TX", "UM", "UT", "VA", "VI", "VT", "WA", "WI", "WV", "WY"));
+
+    public GuestService(GuestRepository repository) {
+        this.repository = repository;
+    }
+
+    public Guest findById(int id){
+        return repository.findById(id);
+    }
+
+    public List<Guest> findByLastName(String prefix) {
+        return repository.findAll().stream()
+                .filter(i -> i.getLastName().startsWith(prefix))
+                .collect(Collectors.toList());
+    }
+
+    public Result<Guest> add(Guest guest) throws DataException {
+        Result<Guest> result = validate(guest);
+        if (!result.isSuccess()){
+            return result;
+        }
+        guest.setPhoneNum(formatPhoneNumber(guest.getPhoneNum()));
+        result.setPayload(repository.add(guest));
+        return result;
+    }
+
+    public Result<Guest> update(Guest guest) throws DataException {
+        Result<Guest> result = validate(guest);
+        if (!result.isSuccess()){
+            return result;
+        }
+        List<Guest> guests = repository.findAll();
+        Guest existing = null;
+        for (Guest g: guests){
+            if (g.getId() == guest.getId()){
+                existing = g;
+            }
+        }
+        if (existing == null){
+            result.addErrorMessage("Guest not found.");
+            return result;
+        }
```

Comment:
The file repository already looks for an existing one to update, can probably skip this check here.